public/js/menu.js

```diff
@@ -1,66 +1,67 @@
-        const queryString = new URLSearchParams(formData).toString();
-        fetch(
-            `https://cmdmethods.onrender.com/api/filter-numbers?${queryString}`
-        )
-            .then((response) => response.json())
-            .then((filterNumbers) => {
-                const list = document.querySelectorAll('.filterNumber');
-                list.forEach((el) => {
-                    if (filterNumbers[el.dataset.filter] != null) {
-                        if (filterNumbers[el.dataset.filter] === 0) {
-                            el.parentElement.previousElementSibling.disabled = true;
-                        } else {
-                            el.parentElement.previousElementSibling.disabled = false;
-                        }
-                        el.textContent = `(${
-                            filterNumbers[el.dataset.filter]
-                        })`;
+    fetch(`/api/filter-numbers?${queryString}`)
+        .then((response) => response.json())
+        .then((filterNumbers) => {
+            const list = document.querySelectorAll(".filterNumber");
+            list.forEach((el) => {
+                if (filterNumbers[el.dataset.filter] != null) {
+                    if (filterNumbers[el.dataset.filter] === 0) {
+                        el.parentElement.previousElementSibling.disabled = true;
+                    } else {
+                        el.parentElement.previousElementSibling.disabled = false;
                     }
-                });
-                document.querySelector(
-                    '#filter-options input[type=submit]'
-                ).value = `Show methods (${filterNumbers.total})`;
-            })
-            .catch((error) => console.log(error));
-    }
+                    el.textContent = `(${filterNumbers[el.dataset.filter]})`;
+                }
+            });
+            document.querySelector(
+                "#filter-options input[type=submit]"
+            ).value = `Show methods (${filterNumbers.total})`;
+        })
+        .catch((error) => console.log(error));
+}
 
-    updateFilterNumbers();
+updateFilterNumbers();
 
-    window.onscroll = function () {
-        const categoryLogos = document.querySelectorAll('.flipable');
-        categoryLogos.forEach((element) => {
-            const observer = new IntersectionObserver(
-                (entries) => {
-                    element.classList.toggle('flip', entries[0].isIntersecting);
-                },
-                { threshold: 1 }
-            );
-            observer.observe(element);
+const observer = new IntersectionObserver(
+    (entries) => {
+        entries.forEach((entry) => {
+            const logo = entry.target.querySelector(".flipable");
+            logo?.classList.toggle("flip", entry.isIntersecting);
         });
-    };
-};
+    },
+    { threshold: 1 }
+);
+
+const categoryLogos = document.querySelectorAll(
+    ".category-description:has(.flipable)"
```

Comment:
Because the individual `.flipable` elements animate in and out, they cause more intersection observer events to fire. To resolve that, we'll watch the intersection of the container of those elements (in this case the `.category-description` box), and then animate the containing logo.